src/picongpu/include/traits/UsesRNG.hpp

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2016 Marco Garten, Rene Widera
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <boost/mpl/bool.hpp>
+
+namespace picongpu
+{
+namespace traits
+{
+
+/** Checks if an object requires the RNG
+ *
+ * @tparam T_Object any object (class or typename)
+ *
+ * This struct must define
+ * ::type (boost::mpl::bool_<>)
+ */
+template<typename T_Object>
+struct UsesRNG
```

Comment:
The advance to use a bool type instead of `bool` value is that it can directly used as operator in any other boot mpl algorithm like `copy_if`, `transform`, ...
Therefore the more generic case is `bool type` instead of `bool value`. 

The best solution would be to use [Tag Dispatched Metafunction](http://www.boost.org/doc/libs/1_53_0/libs/mpl/doc/refmanual/tag-dispatched-metafunction.html)

Than this is allowed:
```C++
if(PMacc::HasFlag<Type, Flag>::value)
```
and also use it as unary functor
```C++
typedef typename boost::mpl::copy_if<LIST,PMacc::HasFlag<boost::mpl::_1, Flag>>::type X;
```>